bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -84,10 +86,10 @@ def initialize(lockfile)
         elsif /^[^\s]/.match?(line)
           @state = nil
         elsif @state
-          send("parse_#{@state}", line)
+          send(PARSE_METHODS.fetch(@state), line)
```

Comment:
👏

This is better code, but what do you think of scrapping `@state` altogether and just storing the `@parser` method name (or keep the name `@state` bit store the method name)?

Personally, I'd save the extra obfuscation and fetching and directly store `@parser = :parse_ruby` etc.

Reply:
yeah that's a good idea